specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
+one content mode, as the recipient is not allowed to negotiate the content mode.
+Senders MUST support both cleartext (`http`) and TLS (`https`) URLs as event
+delivery destinations.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+SHOULD be performed using the CloudEvents HTTP Binding, version 1.0.
+
+Senders MAY probe the recipient with an
+[HTTP OPTIONS request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if
+implemented, the recipient MUST indicate support for the POST verb using the
+[`Allow` header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders
+which receive an error when probing with HTTP OPTIONS SHOULD proceed using the
+HTTP POST mechanism.
+
+### Event Acknowledgement and Delivery Retry
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient SHOULD NOT return a response accepting the event until it
+has handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                           | Retry | Delivery completed | Error |
+| ------------- | --------------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply)       | No    | Yes                | No    |
+| `202`         | Event accepted                    | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `400`         | Unparsable event                  | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist           | Yes   | No                 | Yes   |
+| `409`         | Conflict / Processing in progress | Yes   | No                 | Yes   |
+| `429`         | Too Many Requests / Overloaded    | Yes   | No                 | Yes   |
+| other `4xx`   | Error                             | No    | No                 | Yes   |
+| other `5xx`   | Error                             | Yes   | No                 | Yes   |
+
+\* Unspecified `1xx`, `2xx`, and `3xx` response codes are **reserved for future
+extension**. Event recipients SHOULD NOT send these response codes in this spec
+version, but event senders MUST handle these response codes as errors and
+implement appropriate failure behavior.
+
+Recipients MUST be able to handle duplicate delivery of events (for example, via
+idempotency or tracking event delivery state) and MUST accept delivery of
```

Comment:
I'm not sure what I'd want to say, but it feels like we may need to elaborate on what "accept delivery" means because in some cases it could mean returning a 202 right away. While in other cases it could mean pass it along to the processor, and in others it could mean replay a previously generated response w/o calling the processor. Maybe evening saying that all of these types of things are ok, and that the point of the MUST is to say that it can't reject the request JUST because it's a duplicate.   maybe??

Reply:
yes I like that - thanks

Reply:
did you add this?